masonry/src/widget/textbox.rs

```diff
@@ -170,6 +171,7 @@ impl Widget for Textbox {
                         ctx.request_layout();
                         ctx.request_paint();
                         ctx.request_focus();
+                        ctx.widget_state.needs_window_origin = true;
```

Comment:
I think that moving the `ImeMoved` signal to `paint()` means you can remove this line.